marconi-sidechain/src/Marconi/Sidechain/CLI.hs

```diff
@@ -39,11 +47,21 @@ data CliArgs = CliArgs
   -- synced point.
   , optionsRetryConfig :: !RetryConfig
   }
-  deriving (Show)
+  deriving (Show, Generic, FromJSON, ToJSON)
+
+-- Add orphan instances that aren't provided by Cardano.Api
+deriving instance Generic C.NetworkId
+instance FromJSON C.NetworkMagic
+instance FromJSON C.NetworkId
+instance ToJSON C.NetworkMagic
+instance ToJSON C.NetworkId
```

Comment:
We have a module where we gather all the orphans of Cardano.Api we need, to avoid duplication across packages (`Marconi.ChainIndex.Orphans`), moving these instances here would be great.

Reply:
Thanks! I'll definitely move them there. I was already a little uncomfortable about putting them here in `CLI` anyway.

One thing I think I'll add is to use deriving strategies to be a little more specific about how the instance is being defined. This in turn allows me to use a `newtype` deriving strategy for `NetworkMagic` so that it doesn't have a redundant `unNetworkMagic` in the serialization. In other words, we would have

```json
{
  "networkId": {
    "contents": 1,
    "tag": "Testnet"
  }
}
```

instead of

```json
{
  "networkId": {
    "contents": {
      "unNetworkMagic": 1
    },
    "tag": "Testnet"
  }
}
```